examples/sfu/sfu/media.rs

```diff
@@ -0,0 +1,364 @@
+                let cloned_tx = tx_clone.clone();
+                let cloned_id = uuid.clone();
+
+                Box::pin(async move {
+                    if !candidate.is_none() {
+                        cloned_tx.send(PeerChanCommand::SendIceCandidate {
+                            uuid: cloned_id.to_owned(),
+                            candidate: serde_json::to_string(&candidate.unwrap()).unwrap(),
+                        }).unwrap();
+                    }
+                })
+            })).await;
+
+        self.pc
+            .on_ice_gathering_state_change(Box::new(move |s: RTCIceGathererState| {
+                println!("On ice gathering state change, {}", s);
+                Box::pin(async {})
+            })).await;
+
+        self.pc
+            .on_ice_connection_state_change(Box::new(move |s: RTCIceConnectionState| {
+                println!("On ice connection state change, {}", s);
+                Box::pin(async {})
+            })).await;
+
+
+        self.pc
+            .on_peer_connection_state_change(Box::new(move |s: RTCPeerConnectionState| {
+                println!("Peer Connection State has changed: {}", s);
+                Box::pin(async {})
+            })).await;
+
+        // Set a handler for when a new remote track starts, this handler copies inbound RTP packets,
+        // replaces the SSRC and sends them back
+        let tx_clone = peer_chan_tx.clone();
+        uuid = self.uuid.clone();
+        self.pc
+            .on_track(Box::new(
+                    move |track: Option<Arc<TrackRemote>>, _receiver: Option<Arc<RTCRtpReceiver>>| {
+                        if let Some(track) = track {
+                            tx_clone.send(PeerChanCommand::OnTrack {
+                                uuid: uuid.to_owned(),
+                                track
+                            }).unwrap();
+                        }
+                        Box::pin(async {})
+                    },
+            )).await;
+
+        Ok(())
+    }
+}
+
+const MEDIA: [&'static str; 2] = ["audio", "video"];
+
+// This is ran in a tokio task, that holds all the shared state. It's communicated to by channels.
+pub async fn handle_peer_connection_commands(peer_chan_rx: Receiver<PeerChanCommand>, peer_chan_tx: Sender<PeerChanCommand>) -> Result<()> {
+    let api = crate::sfu::api::prepare_api()?;
+
+    let mut peers: HashMap<String, Peer> = HashMap::new();
+
+    while let Ok(cmd) = peer_chan_rx.recv_async().await {
+        use PeerChanCommand::*;
+
+        match cmd {
+            DistributePacket { uuid, packet, kind } => {
+                match kind {
+                    RTPCodecType::Video => {
+                        for (_k, p) in &peers {
+                            match p.local_video_tracks.get(&uuid) {
+                                Some(track) => {
+                                    if let Err(err) = track.write_rtp(&packet).await {
```

Comment:
These futures can be created, collected into a `Vec` and then ran concurrently with `futures::join_all`